include/deal.II/base/vectorization.h

```diff
@@ -173,15 +173,15 @@ class VectorizedArrayIterator
    * Create new iterator, which is shifted by @p offset.
    */
   VectorizedArrayIterator<T>
-  operator+(const unsigned int &offset) const
+  operator+(const std::size_t &offset) const
   {
     return VectorizedArrayIterator<T>(*data, lane + offset);
   }
 
   /**
    * Compute distance between this iterator and iterator @p other.
    */
-  unsigned int
+  std::size_t
```

Comment:
I don't know the exact semantics of the standard headers out of my head, but shouldn't this be `std::ptrdiff_t` as the difference can get negative?

Reply:
Yes, indeed. This was wrong before already (I didn't pay attention apparently).

Reply:
I have made the change. I was not aware that the value can be also negative. I think the type is not specified and only has to be consistent with `iterator_traits<dealii::VectorizedArrayIterator<T>>::difference_type`